incubator/hnc/internal/validators/object.go

```diff
@@ -146,6 +152,33 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
+// hasConflict checks if there's any conflicting objects in the descendants. Returns
+// true and a list of conflicting descendants, if yes.
+func (o *Object) hasConflict(inst *unstructured.Unstructured) (bool, []string) {
+	o.Forest.Lock()
+	defer o.Forest.Unlock()
+
+	// If the instance is empty (for a delete operation) or it's not namespace-scoped,
+	// there must be no conflict.
+	if inst == nil || inst.GetNamespace() == "" {
```

Comment:
If the instance is marked for deletion but still terminating wont we run into errors when we try to create a different resource of the same name?

Reply:
Isn't that the correct behaviour? We need to wait until the instance is _fully_ deleted?

Reply:
Yes, so then wouldn't we return true (there is a conflict) until it is fully deleted? As is this will indicate there is no conflict (unless I'm reading it wrong).

Reply:
According to #1079 , `Delete` or `Update` an object won't introduce source object overwriting issue so we only discuss the the `Create` operation. This `inst` is the new instance to operate on, and it will only be `nil` for `Delete` operation. That's why it's allowed by the webhook. Does it sound right?

"Fully deleted" concern seems more like for the reconcilers not the webhooks, since if a resource is terminating, the API server won't allow creating a resource with the same name (will report alreadyExist), right?

Reply:
👍

Reply:
@yiqigao217 is right; I was confusing this with a reconciler. Yes, we _never_ want to block a "delete" operation, so this code is correct.